pkg/chunk/purger/purger.go

```diff
@@ -473,7 +477,12 @@ func (p *Purger) pullDeleteRequestsToPlanDeletes() error {
 		p.executePlansChan <- req
 	}
 
-	p.metrics.oldestPendingDeleteRequestAgeSeconds.Set(float64(now.Sub(oldestPendingRequestCreatedAt) / time.Second))
+	// track age of oldest delete request since they are over their cancellation period
+	oldestPendingRequestAge := time.Duration(0)
```

Comment:
Have you considered changing the initial `oldestPendingRequestCreatedAt := now` setting it to `time.Time{}` and then here checking `if !oldestPendingRequestCreatedAt.IsZero()`? I'm mentioning it because I think it's a bit hard to understand why we check `! .Equal(now)` (which is theoretically legit) while you wanna actually check if it was never set.

Reply:
I have used `model.Time(0)` and checked for `0` value instead of `time.Time{}` because `CreatedAt` is of type `model.Time`.